### PR TITLE
Dashbaord lti1.3 navigation bar

### PR DESCRIPTION
**Pull request by IlyaRazuvaev**
https://app.clickup.com/t/86bz4w2ag

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -335,7 +335,15 @@ function local_annoto_extend_settings_navigation(settings_navigation $settingsna
 
     // Check and create LTI external tool.
     require_once($CFG->dirroot . '/mod/lti/locallib.php');
-    $lti = lti_get_tool_by_url_match($settings->toolurl);
+    // Will return all available tools by domain from toolurl.
+    $possibletools = lti_get_tools_by_url($settings->toolurl, LTI_TOOL_STATE_CONFIGURED, null);
```
Comment (genadis) [inline lib.php:339]: does this work for LTI 1.3 without changing ?
image

File: lib.php [hunk 2]
```diff
@@ -335,7 +335,15 @@ function local_annoto_extend_settings_navigation(settings_navigation $settingsna
...
+    $lti = reset($possibletools);
+    foreach ($possibletools as $tool) {
```
Comment (genadis) [inline lib.php:341]: what if there are both LTI 1.3 and LTI 1.1 external tools setup?
There should be a prioritization towards LTI 1.3

File: lib.php (hunk 1 above)
Comment (genadis) [inline lib.php:339]: The approach should be:
1. Get the Annoto LTI tools by using the correct domain.
2. Find LTI 1.3 tool that points to the dashboard (make sure it's not the assignments embed, but the dashboard tool)
3. If not found LTI 1.3, find LTI 1.1

**Review by genadis — CHANGES_REQUESTED** [review]

File: lib.php (hunk 1 above)
Comment (IlyaRazuvaev) [reply lib.php]: Updated to use domain instead

File: lib.php [hunk 3]
```diff
@@ -335,7 +335,25 @@ function local_annoto_extend_settings_navigation(settings_navigation $settingsna
 
     // Check and create LTI external tool.
     require_once($CFG->dirroot . '/mod/lti/locallib.php');
-    $lti = lti_get_tool_by_url_match($settings->toolurl);
+
+    // Will return all available tools by domain from toolurl.
+    $possibletools = lti_get_tools_by_domain($settings->customdomain, LTI_TOOL_STATE_CONFIGURED, null);
```
Comment (genadis) [inline lib.php:340]: for production customdomain will not be used and be empty instead preset domain is used, please fix according to previous similar domain resolution and our discussion

**Review by genadis — CHANGES_REQUESTED** [review]

**Review by genadis — APPROVED** [review]

**Pull request merged**